File: src/kakao/Mbti.java
package kakao;

import java.util.HashMap;
import java.util.Map;

class Mbti {

    public static void main(String[] args) {
        String[] survey = {"AN", "CF", "MJ", "RT", "NA"};
        int[] choices = {5, 3, 2, 7, 5};
        System.out.println(solution(survey, choices));
    }

    public static String solution(String[] survey, int[] choices) {
        StringBuilder mbti = new StringBuilder();

        Map<Character, Integer> map = new HashMap();
        map.put('R', 0);
        map.put('T', 0);
        map.put('C', 0);
        map.put('F', 0);
        map.put('J', 0);
        map.put('M', 0);
        map.put('A', 0);
        map.put('N', 0);

        for (int i = 0; i < survey.length; i++) {
            char[] chars = {survey[i].charAt(0), survey[i].charAt(1)};
            int score = (choices[i] - 4);
            if(score> 0) {
                map.replace(chars[1], map.get(chars[1]) + score);
            } else {
                map.replace(chars[0], map.get(chars[0]) + Math.abs(score));
            }
        }

        if(map.get('R') >= map.get('T')) {
            mbti.append('R');
        } else {
            mbti.append('T');
        }
        if(map.get('C') >= map.get('F')) {
            mbti.append('C');
        } else {
            mbti.append('F');
        }
        if(map.get('J') >= map.get('M')) {
            mbti.append('J');
        } else {
            mbti.append('M');
        }
        if(map.get('A') >= map.get('N')) {
            mbti.append('A');
        } else {
            mbti.append('N');
        }


        return mbti.toString();
    }

}

File: src/kakao/Mbti.kt
package kakao

import java.lang.StringBuilder
import kotlin.math.abs

fun main() {
    val survey = arrayOf("AN", "CF", "MJ", "RT", "NA")
    val choices = intArrayOf(5, 3, 2, 7, 5)
//    val s = Solution(3, );
//    println( s.solution(survey, choices) )
}
class Solution(n: Int, arr: Array<IntArray>) {
    fun solution(survey: Array<String>, choices: IntArray): String {
        var answer = StringBuilder()
        val map = mutableMapOf<Char, Int>(
            'R' to 0,
            'T' to 0,
            'C' to 0,
            'F' to 0,
            'J' to 0,
            'M' to 0,
            'A' to 0,
            'N' to 0,
        )

        for (i in survey.indices) {
            val chars: CharArray = survey[i].toCharArray()
            val score = choices[i] -4
            if(score > 0) {
                map.replace(chars[1], map[chars[1]]!! + score)
            } else {
                map.replace(chars[0], map[chars[0]]!! + abs(score))
            }
        }
        if(map['T']!! > map['R']!!) {
            answer.append('T')
        } else {
            answer.append('R')
        }
        if(map['F']!! > map['C']!!) {
            answer.append('F')
        } else {
            answer.append('C')
        }
        if(map['M']!! > map['J']!!) {
            answer.append('M')
        } else {
            answer.append('J')
        }
        if(map['N']!! > map['A']!!) {
            answer.append('N')
        } else {
            answer.append('A')
        }

        return answer.toString()
    }
}